### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by cs-yash-patodkar**
Initial code for top X customers

Description:
The widget displays 'top 3' from the module selected in the edit config, grouped by either picklist or a lookup field

PMDB : 24563

### CODE REVIEW

File: widget/assets/css/top3.css [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.layer-border-left {
+    border-radius: 4px;
+    height: 80px;
+    width: 350px;
+    margin-top: 20px;
```
Comment (cs-shabaz-salmani) [inline widget/assets/css/top3.css]: please use pre-defined class for margin, padding, display

for margin-top: 20px use margin-top-20 class
for display: block use display-block class
for margin-left use margin-left-md class

this way we can save number of lines of repetitive code

File: widget/view.controller.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        }
+        init();
+
+        function getObjectById(data, name) {
+            for (let i = 0; i < data.length; i++) {
+                if (data[i].name === name) {
+                    if (data[i].value === "true") {
+                        return true;
+                    }
+                    else {
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js]: You can remove the else stmt and just return false

File: widget/view.controller.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                        return false;
+                    }
+                }
+            }
+            return false; // return false if no object with the given id is found
+        }
+
+        function eventListner(){
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js]: Need to remove this code for 1.0.0

File: widget/view.controller.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            pagedTotalData.loadByPost(filters).then(function () {
+                var data = pagedTotalData.fieldRows[0][$scope.config.customModuleField].value;
+                var nestedKeysArray = $scope.config.keyForCustomModule.split('.');
+                if (nestedKeysArray.length >= 1) {
+                    nestedKeysArray.forEach(function (value) {
+                        data = data[value];
+                    })
+                }
+                if (data === undefined) {
+                    _dataSource = { "Key is invalid... ": "" }
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js]: Invalid Key

**Review by cs-suyog-jinturkar — COMMENTED** [review]
Added few comments

File: widget/info.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    "title": "Top 3",
+    "subTitle": "Top 3 Widget",
+    "version": "1.0.0",
+    "metadata": {
+        "description": "",
+        "pages": [
+            "Dashboard"
+        ],
+        "certified": "No",
+        "publisher": "Community",
```
Comment (cs-pranali-chavan) [inline widget/info.json:12]: "certified": "Yes",
"publisher": "Fortinet",
description required
"help_online":"" with read me file path required

File: widget/info.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "top3",
+    "title": "Top 3",
+    "subTitle": "Top 3 Widget",
+    "version": "1.0.0",
```
Comment (cs-pranali-chavan) [inline widget/info.json:5]: "published_date" missing

**Review by cs-pranali-chavan — COMMENTED** [review]
minor changes in info.json 
rest of the code seems good.

**Pull request closed without merging**